### PR TITLE
Add basic e2e for NGM

### PR DESCRIPTION
**Pull request by KanShiori**
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: tests/actions.go [hunk 1]
```diff
@@ -3747,6 +3748,78 @@ func (oa *OperatorActions) WaitForDmClusterDeleted(ns, dcName string, timeout, p
...
+		log.Logf("TiDBNGMonitoring NGM Sts %q: CurrentRevision(%s) != UpdateRevision(%s)", stsLocator, sts.Status.CurrentRevision, sts.Status.UpdateRevision)
+		return false
+	}
+
+	if !utilstatefulset.IsAllDesiredPodsRunningAndReady(helper.NewHijackClient(oa.kubeCli, oa.asCli), sts) {
+		return false
+	}
+
+	if tngm.Status.NGMonitoring.StatefulSet == nil {
+		log.Logf("TiDBNGMonitoring NGM %q: .Status.Master.StatefulSet is nil", tngmLocator)
```
Comment (csuzhangxc) [inline tests/actions.go:3804]:
```suggestion
		log.Logf("TiDBNGMonitoring NGM %q: .Status.NGMonitoring.StatefulSet is nil", tngmLocator)
```

File: tests/e2e/tidbngmonitoring/ngm.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+			tc := GetTCForTiDBNGMonitoring(ns, name)
+			tngm := fixture.GetTidbNGMonitoring(ns, name, tc)
+
+			ginkgo.By("Deploy tidb cluster")
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 5*time.Minute, 10*time.Second)
+
+			ginkgo.By("Deploy tidb ng monitoring")
+			err := genericCli.Create(context.TODO(), tngm)
+			framework.ExpectNoError(err, "failed to create TidbNGMonitoring %s", locator)
+			err = oa.WaitForTiDBNGMonitoringReady(tngm, 1*time.Minute, 10*time.Second)
```
Comment (csuzhangxc) [inline tests/e2e/tidbngmonitoring/ngm.go:119]: Is 1 minute enough?

File: tests/e2e/tidbngmonitoring/ngm.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+			ginkgo.By("Installing initial tidb CA certificate")
+			err := tidbcluster.InstallTiDBIssuer(ns, name)
+			framework.ExpectNoError(err, "failed to install CA certificate")
+
+			ginkgo.By("Install tidb cluster certificate")
+			framework.ExpectNoError(tidbcluster.InstallTiDBComponentsCertificates(ns, name), "failed to install TiDB server and client certificate")
+			<-time.After(30 * time.Second)
+
+			ginkgo.By("Deploy tidb cluster with TLS enabled")
+			tc.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
```
Comment (csuzhangxc) [inline tests/e2e/tidbngmonitoring/ngm.go:152]: Is it needed to enable TLS for client?

**Review by csuzhangxc — APPROVED** [review]

File: tests/e2e/tidbngmonitoring/ngm.go (hunk 3 above)
Comment (DanielZhangQD) [reply tests/e2e/tidbngmonitoring/ngm.go:152]: Yes, it's required.

**Pull request merged**